tools/pubsys/src/aws/publish_ami/mod.rs

```diff
@@ -150,21 +147,21 @@ pub(crate) async fn run(args: &Args, publish_args: &PublishArgs) -> Result<()> {
     info!("Waiting for AMIs to be available...");
     let mut wait_requests = Vec::with_capacity(amis.len());
     for (region, image) in &amis {
-        let wait_future = wait_for_ami(&image.id, region, &base_region, "available", 1, &aws);
+        let wait_future = wait_for_ami(&image.id, region, base_region, "available", 1, &aws);
         // Store the region and ID so we can include it in errors
         let info_future = ready((region.clone(), image.id.clone()));
         wait_requests.push(join(info_future, wait_future));
     }
     // Send requests in parallel and wait for responses, collecting results into a list.
     let request_stream = stream::iter(wait_requests).buffer_unordered(4);
-    let wait_responses: Vec<((Region, String), std::result::Result<(), wait::Error>)> =
+    let wait_responses: Vec<((String, String), std::result::Result<(), wait::Error>)> =
         request_stream.collect().await;
 
     // Make sure waits succeeded and AMIs are available.
     for ((region, image_id), wait_response) in wait_responses {
         wait_response.context(error::WaitAmiSnafu {
             id: &image_id,
-            region: region.name(),
+            region: &region,
```

Comment:
nit: can we do this?
```suggestion
            region,
```